### ISSUE TITLE
Statements surrounded by brackets still parse

### ISSUE THREAD

**Issue by jpy-git** [sqlfluff/sqlfluff#1899]
If a statement is wrapped in brackets I would expect it to fail to parse. However, the statement still parses fine and throws no error.
e.g. 
```SQL
(SELECT foo
FROM bar)
```

## Expected Behaviour
Parse error as invalid SQL.

## Observed Behaviour
![Screen-Recording-2021-11-13-at-22 47 45](https://user-images.githubusercontent.com/80432516/141661478-acaa657a-5a30-4aa7-b415-452db4b5d786.gif)

## Steps to Reproduce
Lint the example query. It also parses INSERT/UPDATE/WITH statements wrapped in brackets.

## Dialect
ANSI (although issue is the same regardless of selected dialect)

## Version
Include the output of `sqlfluff --version` along with your Python version
`python 3.9.8`
`sqlfluff, version 0.8.1` (HEAD of main branch)

## Configuration
Default configuration.

**Comment by barrywhart** [comment]
I did a quick test with SQLite to confirm that this is flagged as a parse error by the database. (I was initially wondering if this might be valid syntax, since parentheses around a query are often used with nested queries.)

The transcript below has two examples -- the first with a parse error, and the second with a nested query, where the parentheses are legal.
```
$ sqlite3
SQLite version 3.32.3 2020-06-18 14:16:19
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> create table foo (int i);
sqlite> (select * from foo);
Error: near "(": syntax error
sqlite> select * from (select * from foo);
```

**Comment by mattssll** [comment]
it's not letting single queries be inside parenthesis, this should be allowed as it's not a syntax error, it could be a styling/formatting issue perhaps.

**Comment by jpy-git** [comment]
Okay so when I spotted this I was using sqlite as @barrywhart did and we can see that this is a syntax error in that dialect.

I have now tried the same query on Snowflake/MySQL and it seems to not be a syntax error for those dialects so maybe it's not an issue that this is parsing. However, we don't specify OptionallyBracketed around lone select clauses in the grammar, so it would be good to understand if this is intentional behaviour of the parser?

> it's not letting single queries be inside parenthesis

^ @mattssll Not sure what you mean by this, single queries inside brackets are currently being allowed by sqlfluff and nothing has been changed to stop that. 

I agree the sensible approach would be to make a rule that can flag/remove brackets around lone queries, and then people can still disable it if they (for some reason) want those brackets.

I'll pull together a PR
